src/main/java/net/openhft/chronicle/core/io/QueryCloseable.java

```diff
@@ -5,10 +5,23 @@
 package net.openhft.chronicle.core.io;
 
 public interface QueryCloseable {
+
+    /**
+     * Is this object closed, or in the process of closing
+     * <p>
+     * This should always be true if {@link #isClosed()} is true
```

Comment:
This sounds wrong: if the closeable is "closed" it is no longer "closing"
So isClosing() should not always be true if isClosed() is true
Can you add a state diagram to illustrate pls

Reply:
Hi Roger, apologies, I've written a proper description now, please see above. I'm not proposing a change to behaviour, but instead attempting to clarify the contract provided by the QueryCloseable interface.